data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/node/GetPeerCount.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.handlers.v1.node;
+
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.CACHE_NONE;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_INTERNAL_ERROR;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.RES_OK;
+import static tech.pegasys.teku.beaconrestapi.RestApiConstants.TAG_NODE;
+
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.teku.api.DataProvider;
+import tech.pegasys.teku.api.NetworkDataProvider;
+import tech.pegasys.teku.api.response.v1.node.GetPeerCountResponse;
+import tech.pegasys.teku.api.response.v1.node.Peer;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class GetPeerCount implements Handler {
+  public static final String ROUTE = "/eth/v1/node/peer_count";
+  private final JsonProvider jsonProvider;
+  private final NetworkDataProvider network;
+
+  public GetPeerCount(final DataProvider provider, final JsonProvider jsonProvider) {
+    this.jsonProvider = jsonProvider;
+    this.network = provider.getNetworkDataProvider();
+  }
+
+  GetPeerCount(final NetworkDataProvider network, final JsonProvider jsonProvider) {
+    this.network = network;
+    this.jsonProvider = jsonProvider;
+  }
+
+  @OpenApi(
+      path = ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get peer count",
+      tags = {TAG_NODE},
+      description = "Retrieves number of known peers.",
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetPeerCountResponse.class)),
+        @OpenApiResponse(status = RES_INTERNAL_ERROR)
+      })
+  @Override
+  public void handle(@NotNull final Context ctx) throws Exception {
+    ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
+    List<Peer> peers = network.getPeers();
```

Comment:
We could potentially improve this API and the get peers API by adding a method to `Eth2P2PNetwork` to `streamAllPeers` which doesn't filter for peers that are ready.  Then we'd at least be able to report `connecting` peers (where `peer.hasStatus()` is false).

Not sure it's worth it though.

Reply:
It might be good to get the information into the peers list, I'm not sure how the information is used though, and whether it's worth the effort...